backend/schema/validate.go

```diff
@@ -613,20 +643,72 @@ func validateVerbMetadata(scopes Scopes, module *Module, n *Verb) (merr []error)
 	return
 }
 
-func validateIngressRequestOrResponse(scopes Scopes, module *Module, n *Verb, reqOrResp string, r Type) (fieldType Type, body Symbol, merr []error) {
+type HTTPRequestExtractedTypes struct {
+	fieldType        Type
+	body             Symbol
+	pathParamType    Type
+	pathParamSymbol  Symbol
+	queryParamType   Type
+	queryParamSymbol Symbol
+}
+
+func validateIngressResponse(scopes Scopes, module *Module, n *Verb, reqOrResp string, r Type) (merr []error) {
```

Comment:
Could `reqOrResp` be a bool or enum?

Reply:
It's mostly used for constructing error messages, it was part of the existing code.